runtime/sema/type.go

```diff
@@ -4033,6 +4034,23 @@ func isAttachmentType(t Type) bool {
 		t == AnyStructAttachmentType
 }
 
+func IsHashableStructType(t Type) bool {
+	switch typ := t.(type) {
+	case *AddressType:
+		return true
+	case *CompositeType:
+		return typ.Kind == common.CompositeKindEnum
+	default:
+		switch typ {
+		case NeverType, BoolType, CharacterType, StringType, MetaType, HashableStructType:
```

Comment:
This is mostly same as the existing `IsValidDictionaryKeyType` function but I've added it to also return `true` for `HashableStructType` i.e. `IsHashableStructType(HashableStructType) == true`.

Reply:
Can you add this as a comment in the code too?